File: src/main/org/nlogo/prim/etc/_clearall.java
package org.nlogo.prim.etc;

import org.nlogo.api.LogoException;
import org.nlogo.api.Syntax;

public final strictfp class _clearall
    extends org.nlogo.nvm.Command {
  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax("O---", true);
  }

  @Override
  public void perform(final org.nlogo.nvm.Context context) throws LogoException {
    perform_1(context);
  }

  public void perform_1(final org.nlogo.nvm.Context context) throws LogoException {
    workspace.clearAll();
    context.ip = next;
  }
}

File: src/main/org/nlogo/prim/etc/_clearall.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Command, Context }

class _clearall extends Command {
  override def syntax =
    Syntax.commandSyntax("O---", true)
  override def perform(context: Context) {
    workspace.clearAll()
    context.ip = next
  }
}
